netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -607,7 +644,27 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
           logger.log(Level.WARNING, String.format("Received unexpected ping ack. "
               + "Expecting %d, got %d", p.payload(), ackPayload));
         }
-      } else {
+      }
+      else if (data.readLong() == 1234){     
```

Comment:
Where do these numbers come from?  IIRC we allow user code to send pings too.  How do you tell your pings from theirs?